### ISSUE TITLE
ImGuiColorEditFlags_NoTooltip does not work in conjunction with ImGuiColorEditFlags_NoSidePreview

### ISSUE THREAD

**Issue by ambiennt** [ocornut/imgui#6957]
**Version/Branch of Dear ImGui:**

Version: 1.89.9
Branch: shadows

**Back-end/Renderer/Compiler/OS**
Back-ends: imgui_impl_win32.cpp + imgui_impl_dx11.cpp
Compiler:  MSVC/Clang-cl
Operating System: Windows 10

**My Issue/Question:**
When using a color picker widget, such as `ImGui::ColorPicker4`, setting the ImGuiColorEditFlags_NoTooltip only removes the color preview tooltip when ImGuiColorEditFlags_NoSidePreview is NOT set. Otherwise, the color preview still has a tooltip for the small color squares to the right of the RGBA/HSVA/hex input cells.

**Standalone, minimal, complete and verifiable example:**
*(in both screenshots, my mouse is hovered over the black square, but my screenshot tool wasn't capturing it)*
```
ImGui::Begin("Example Bug");
static ImVec4 col{};
// works as expected, larger color preview still shows on the righthand side and hovering over it with mouse has no effect
ImGui::ColorPicker4("My Color Picker", (float*)&col, ImGuiColorEditFlags_NoTooltip);
ImGui::End();
```
![case1](https://github.com/ocornut/imgui/assets/63216972/0c077f47-0a26-41e5-9da9-37a088611196)
<br>
```
ImGui::Begin("Example Bug");
static ImVec4 col{};
// disables the color preview cell on the righthand side, and instead places the previews to the right of the color input cells
// ImGuiColorEditFlags_NoTooltip has no effect as the tooltip still shows when hovering over these
ImGui::ColorPicker4("My Color Picker", (float*)&col, ImGuiColorEditFlags_NoTooltip | ImGuiColorEditFlags_NoSidePreview);
ImGui::End();
```
![case2](https://github.com/ocornut/imgui/assets/63216972/1802a802-dde7-4dba-ad6e-95795418f11b)
<br>
I expect the ImGuiColorEditFlags_NoTooltip flag to hide tooltips for both kinds of previews, contrary to the observed behavior.

**Comment by ocornut** [comment]
Thank you for the careful report and repro code.

If I have to be honest I would say that `ImGuiColorEditFlags_NoTooltip` is a little bit underdefined for `ColorPicker4()` : our codebase only ever used it when calling `ColorEdit4()`.

(1) In the code that does:
```cpp
    // R,G,B and H,S,V slider color editor
    bool value_changed_fix_hue_wrap = false;
    if ((flags & ImGuiColorEditFlags_NoInputs) == 0)
    {
        PushItemWidth((alpha_bar ? bar1_pos_x : bar0_pos_x) + bars_width - picker_pos.x);
        ImGuiColorEditFlags sub_flags_to_forward = ImGuiColorEditFlags_DataTypeMask_ | ImGuiColorEditFlags_InputMask_ | ImGuiColorEditFlags_HDR | ImGuiColorEditFlags_NoAlpha | ImGuiColorEditFlags_NoOptions | ImGuiColorEditFlags_NoSmallPreview | ImGuiColorEditFlags_AlphaPreview | ImGuiColorEditFlags_AlphaPreviewHalf;
        ImGuiColorEditFlags sub_flags = (flags & sub_flags_to_forward) | ImGuiColorEditFlags_NoPicker;
```
I believe we need to add `ImGuiColorEditFlags_NoTooltip` to `sub_flags_to_forward`.

(2) However, I would argue the source issue that `ColorPicker4()` only conditionally set `ImGuiColorEditFlags_NoSmallPreview`:
```
    if (!(flags & ImGuiColorEditFlags_NoSidePreview))
        flags |= ImGuiColorEditFlags_NoSmallPreview;
```

The original commit which added this behavior is 3926bd08e which is ANCIENT.
I don't think it's necessarily a desirable behavior that we perform this conditional, but I suppose my reasoning at the time was:
- if you don't have a SidePreview you are more likely to want to a preview SOMEWHERE and those small previews next to individual DragFloat3 widgets would make sense.
- you can always opt-out manually by adding `ImGuiColorEditFlags_NoSmallPreview`, whereas if we made this the default we wouldn't be able to opt-out.

So even though I am not super happy with this, I still reckon the current behavior may be senseful.

I would therefore only fix point (1): pushed 88fec09.

Let me know if you agree with my reasoning on (2) !

**Comment by ambiennt** [comment]
Thank you for the timely hotfix, I do believe [88fec09](https://github.com/ocornut/imgui/commit/88fec097158867edf7753fe0d842d23d67e7966d) sufficiently addresses the issue.

As for (2), I agree with your reasoning because I believe the default expectation should be that a preview exists somewhere. Similarly, I also believe that it should be on the user to go out of their way to disable it if they really want. 

Although (unless I missed it) I think maybe there could be some clarification in the form of a comment outlining how the default flags work. At least for me, it was a lot of guesswork trying different combinations of flags and comparing against visual observations to understand how some of the overriding behavior worked.
